avocado/core/sysinfo.py

```diff
@@ -161,8 +162,24 @@ def run(self, logdir):
         stdin = open(os.devnull, "r")
         stdout = open(logf_path, "w")
         try:
-            subprocess.call(self.cmd, stdin=stdin, stdout=stdout,
-                            stderr=subprocess.STDOUT, shell=True, env=env)
+            proc = subprocess.Popen(self.cmd, stdin=stdin, stdout=stdout,
+                                    stderr=subprocess.STDOUT, shell=True,
+                                    env=env)
+            deadline = time.time() + 10
+            while time.time() < deadline:
+                if proc.poll() is not None:
+                    break
+                time.sleep(0.01)
```

Comment:
This will be busy polling the process, making the CPU usage to go up if the process spends a long time to finish (our main corner case we want to fix). So what about implementing something similar as yourself did on 1bf7163fc21e23ef82d703a0fe25eac49d6b9e94?